### ISSUE TITLE
Warning for uninitialized field with [AllowNull]

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#39921]
No uninitialized field warnings should be reported for the following:
```C#
#nullable enable

using System.Diagnostics.CodeAnalysis;

class C<T>
{
    [AllowNull]T F1;            // warning CS8618: Non-nullable field 'F1' is uninitialized
    [MaybeNull, AllowNull]T F2; // warning CS8618: Non-nullable field 'F2' is uninitialized
}
```
cc @stephentoub

**Comment by cston** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/37138.
